### PR TITLE
Release 18.0 Snapshot

### PR DESCRIPTION
**Pull request by palnabarun**
- [x] Update submodule
- [x] Update constants
- [x] Generate client
- [x] Update CHANGELOG

TODO After Merge:
- [ ] Cut alpha release for v18
- [ ] Cherry-pick CHANGELOG and README into `master`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,36 @@
+# v18.0.0-snapshot
+
+Kubernetes API Version: 1.18.12
+
+**Important Information:**
+
+- The Kubernetes Python client versioning scheme has changed. The version numbers used till Kubernetes Python Client v12.y.z lagged behind the actual Kubernetes minor version numbers. From this release, the client is moving a version format `vY.Z.P` where `Y` and `Z` are respectively from the Kubernetes version `v1.Y.Z` and `P` would incremented due to changes on the Python client side itself. Ref: https://github.com/kubernetes-client/python/issues/1244
+- Python 2 had reached [End of Life](https://www.python.org/doc/sunset-python-2/) on January 1, 2020. The Kubernetes Python Client has dropped support for Python 2 from this release (v18.0.0) and will no longer provide support to older clients as per the [Kubernetes support policy](https://kubernetes.io/docs/setup/release/version-skew-policy/#supported-versions).
+
+**API Change:**
```
Comment (roycaihw) [inline CHANGELOG.md:10]: Some major API change in "Generate client 18.0.0-snapshot":
```
kubernetes/client/api/apps_v1beta1_api.py                                          |  5658 ---
kubernetes/client/api/apps_v1beta2_api.py                                          |  9234 -----
kubernetes/client/api/extensions_v1beta1_api.py                                    |  9411 +----
```
Could you add a note about these APIs being removed from this client? 
```
- the following deprecated APIs can no longer be served:
  - All resources under `apps/v1beta1` and `apps/v1beta2` - use `apps/v1` instead
  - `daemonsets`, `deployments`, `replicasets` resources under `extensions/v1beta1` - use `apps/v1` instead
  - `networkpolicies` resources under `extensions/v1beta1` - use `networking.k8s.io/v1` instead
  - `podsecuritypolicies` resources under `extensions/v1beta1` - use `policy/v1beta1` instead ([#85903](https://github.com/kubernetes/kubernetes/pull/85903), [@liggitt](https://github.com/liggitt)) [SIG API Machinery, Apps, Cluster Lifecycle, Instrumentation and Testing]
```

**Comment by roycaihw** [comment]
LGTM overall. Thanks for putting the effort collecting the release notes. 

It surprises me that we have way more changes in 1.18 (+66,885 −136,266) than in [1.17](https://github.com/kubernetes-client/python/pull/1307) (+22,657 −4,082). It looks like the openapi spec itself changed a lot. 
```
kubernetes/swagger.json.unprocessed                                                | 82610 +++++++++++++++------------------
scripts/swagger.json                                                               | 40354 ++++++------------
```
It could be because the upstream dropped some deprecated APIs. Could you confirm that?

File: CHANGELOG.md (hunk 1 above)
Comment (palnabarun) [reply CHANGELOG.md:17]: Thanks for reminding me. I had forgotten to put the removal notices.

**Comment by palnabarun** [comment]
> It could be because the upstream dropped some deprecated APIs. Could you confirm that?

It may be due to the fact that `release-18.0` doesn't have the latest changes in `master` and this PR also adds those changes in the `release-18.0`. This PR is ideally just 10 commits. The other 13 commits are updating the branch itself. We can split those changes into two different PRs if that makes sense.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,36 @@
...
+- Fix bug in reflector that couldn't recover from "Too large resource version" errors ([#92537](https://github.com/kubernetes/kubernetes/pull/92537), [@wojtek-t](https://github.com/wojtek-t)) [SIG API Machinery]
```
[the remark this answers is not in the document]
Comment (palnabarun) [reply CHANGELOG.md:18]: Yeah. I kept it in the changelogs for the API server error exposure. Do you think adding a few more lines with the upstream CHANGELOG add value to this?

[the remark this answers is not in the document]
Comment (roycaihw) [reply CHANGELOG.md:18]: Sounds good. Thanks

**Pull request merged**